doc/v5.0/man3/MPI_Comm_dup_with_info.3
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "MPI_Comm_dup_with_info" "3" "" "2021-08-30" "Open MPI"
.hy
.SH Name
.PP
\f[C]MPI_Comm_dup_with_info\f[R] - Duplicates an existing communicator
using provided info.
.SH Syntax
.SS C Syntax
.IP
.nf
\f[C]
#include <mpi.h>

int MPI_Comm_dup_with_info(MPI_Comm comm, MPI_Info info, MPI_Comm *newcomm)
\f[R]
.fi
.SS Fortran Syntax
.IP
.nf
\f[C]
USE MPI
! or the older form: INCLUDE \[aq]mpif.h\[aq]

MPI_COMM_DUP_WITH_INFO(COMM, INFO, NEWCOMM, IERROR)
    INTEGER COMM, INFO, NEWCOMM, IERROR
\f[R]
.fi
.SS Fortran 2008 Syntax
.IP
.nf
\f[C]
USE mpi_f08

MPI_Comm_dup_with_info(comm, info, newcomm, ierror)
    TYPE(MPI_Comm), INTENT(IN) :: comm
    TYPE(MPI_Info), INTENT(IN) :: info
    TYPE(MPI_Comm), INTENT(OUT) :: newcomm
    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
\f[R]
.fi
.SH Input Parameter
.IP \[bu] 2
\f[C]comm\f[R] : Communicator (handle).
.IP \[bu] 2
\f[C]info\f[R] : Info argument (handle).
.SH Output Parameters
.IP \[bu] 2
\f[C]newcomm\f[R] : Copy of comm (handle).
.IP \[bu] 2
\f[C]IERROR\f[R] : Fortran only: Error status (integer).
.SH Description
.PP
\f[C]MPI_Comm_dup_with_info\f[R] acts exactly like
\f[C]MPI_Comm_dup\f[R] except that the \f[C]info\f[R] hints associated
with the communicator \f[C]comm\f[R] are not duplicated in
\f[C]newcomm\f[R].
The hints provided by the argument \f[C]info\f[R] are associated with
the output communicator \f[C]newcomm\f[R] instead.
See \f[C]MPI_Comm_set_info(3)\f[R] for the list of recognized
\f[C]info\f[R] keys.
.SH Notes
.PP
This operation is used to provide a parallel library call with a
duplicate communication space that has the same properties as the
original communicator.
This includes any attributes (see below) and topologies (see Chapter 6,
\[lq]Process Topologies,\[rq] in the MPI-1 Standard).
This call is valid even if there are pending point-to-point
communications involving the communicator \f[C]comm\f[R].
A typical call might involve an \f[C]MPI_Comm_dup_with_info\f[R] at the
beginning of the parallel call, and an \f[C]MPI_Comm_free\f[R] of that
duplicated communicator at the end of the call.
Other models of communicator management are also possible.
This call applies to both intra- and intercommunicators.
Note that it is not defined by the MPI standard what happens if the
attribute copy callback invokes other MPI functions.
In Open MPI, it is not valid for attribute copy callbacks (or any of
their children) to add or delete attributes on the same object on which
the attribute copy callback is being invoked.
.SH Errors
.PP
Almost all MPI routines return an error value; C routines as the value
of the function and Fortran routines in the last argument.
Before the error value is returned, the current MPI error handler is
called.
By default, this error handler aborts the MPI job, except for I/O
function errors.
The error handler may be changed with \f[C]MPI_Comm_set_errhandler\f[R];
the predefined error handler \f[C]MPI_ERRORS_RETURN\f[R] may be used to
cause error values to be returned.
Note that MPI does not guarantee that an MPI program can continue past
an error.
.SH See Also
.PP
\f[C]MPI_Comm_dup\f[R](3) \f[C]MPI_Comm_idup\f[R](3)
\f[C]MPI_Comm_set_info\f[R](3)


doc/v2.0/man3/MPI_Comm_dup_with_info.3.php
<?php
$topdir = "../../..";
$title = "MPI_Comm_dup_with_info(3) man page (version 2.0.4)";
$meta_desc = "Open MPI v2.0.4 man page: MPI_COMM_DUP_WITH_INFO(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
      <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Comm_dup_with_info </b> -  Duplicates an existing communicator
using provided info.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Comm_dup_with_info(MPI_Comm comm, MPI_Info info, MPI_Comm *newcomm)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_COMM_DUP_WITH_INFO(COMM, INFO, NEWCOMM, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;COMM, INFO, NEWCOMM, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>Input Parameter</a></h2>

<dl>

<dt>comm </dt>
<dd>Communicator (handle). </dd>

<dt>info </dt>
<dd>Info argument (handle).
<p>
</dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameters</a></h2>

<dl>

<dt>newcomm </dt>
<dd>Copy of comm (handle). </dd>

<dt>IERROR </dt>
<dd>Fortran only: Error
status (integer).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
MPI_Comm_dup_with_info acts exactly like <a href="../man3/MPI_Comm_dup.3.php">MPI_Comm_dup</a>
except that the info hints associated with the communicator <i>comm</i> are not
duplicated in <i>newcomm</i>. The hints provided by the argument <i>info</i> are associated
with the output communicator <i>newcomm</i> instead.
<p>
<h2><a name='sect7' href='#toc7'>Notes</a></h2>
This operation is used
to provide a parallel library call with a duplicate communication space
that has the same properties as the original communicator. This includes
any attributes (see below) and topologies (see Chapter 6, "Process Topologies,"
in the MPI-1 Standard). This call is valid even if there are pending point-to-point
communications involving the communicator comm. A typical call might involve
an MPI_Comm_dup_with_info at the beginning of the parallel call, and an
<a href="../man3/MPI_Comm_free.3.php">MPI_Comm_free</a> of that duplicated communicator at the end of the call. Other
models of communicator management are also possible. <p>
This call applies to
both intra- and intercommunicators.
<p> Note that it is not defined by the MPI
standard what happens if the attribute copy callback invokes other MPI
functions.  In Open MPI, it is not valid for attribute copy callbacks (or
any of their children) to add or delete attributes on the same object on
which the attribute copy callback is being invoked.
<p>
<h2><a name='sect8' href='#toc8'>Errors</a></h2>
Almost all MPI
routines return an error value; C routines as the value of the function
and Fortran routines in the last argument. <p>
Before the error value is returned,
the current MPI error handler is called. By default, this error handler
aborts the MPI job, except for I/O function errors. The error handler may
be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the predefined error handler MPI_ERRORS_RETURN
may be used to cause error values to be returned. Note that MPI does not
guarantee that an MPI program can continue past an error.
<p>
<h2><a name='sect9' href='#toc9'>See Also</a></h2>
<a href="../man3/MPI_Comm_dup.3.php">MPI_Comm_dup</a>
<a href="../man3/MPI_Comm_idup.3.php">MPI_Comm_idup</a> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input Parameter</a></li>
<li><a name='toc5' href='#sect5'>Output Parameters</a></li>
<li><a name='toc6' href='#sect6'>Description</a></li>
<li><a name='toc7' href='#sect7'>Notes</a></li>
<li><a name='toc8' href='#sect8'>Errors</a></li>
<li><a name='toc9' href='#sect9'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
